### PR TITLE
Update Logtalk lexer for the latest language specification

### PR DESCRIPTION
**Pull request by pmoura**
Tested using the sample Logtalk source file available at:
https://github.com/LogtalkDotOrg/logtalk3/tree/master/coding/tests

### CODE REVIEW

File: pygments/lexers/prolog.py [hunk 1]
```diff
@@ -226,16 +228,16 @@ class LogtalkLexer(RegexLexer):
...
-            (r"[a-z]\w*", Text),
-            (r"'", String, 'quoted_atom'),
+            (r"[a-z][a-zA-Z0-9_]*", Text),
+            (r"[']", String, 'quoted_atom'),
         ],
 
         'quoted_atom': [
-            (r"''", String),
-            (r"'", String, '#pop'),
+            (r"['][']", String),
```
Comment (Anteru) [inline pygments/lexers/prolog.py]: Can you please shed some light on this? Why is a character class selector?

File: pygments/lexers/prolog.py [hunk 2]
```diff
@@ -226,16 +228,16 @@ class LogtalkLexer(RegexLexer):
             (r'\^', Operator),
             # Strings
             (r'"(\\\\|\\"|[^"])*"', String),
-            # Ponctuation
+            # Punctuation
             (r'[()\[\],.|]', Text),
             # Atoms
-            (r"[a-z]\w*", Text),
-            (r"'", String, 'quoted_atom'),
+            (r"[a-z][a-zA-Z0-9_]*", Text),
```
Comment (Anteru) [inline pygments/lexers/prolog.py]: ` \w` should be equivalent? Why this change?

File: pygments/lexers/prolog.py [hunk 3]
```diff
@@ -244,44 +246,43 @@ class LogtalkLexer(RegexLexer):
...
-            (r'(alias|d(ynamic|iscontiguous)|m(eta_(non_terminal|predicate)|ode|ultifile)|'
-             r's(et_(logtalk|prolog)_flag|ynchronized))(?=[(])', Keyword, 'root'),
+            (r'(built_in|dynamic|synchronized|threaded)(?=[.])', Keyword, 'root'),
+            (r'(alias|d(ynamic|iscontiguous)|m(eta_(non_terminal|predicate)|ode|ultifile)|s(et_(logtalk|prolog)_flag|ynchronized))(?=[(])', Keyword, 'root'),
             (r'op(?=[(])', Keyword, 'root'),
             (r'(c(alls|oinductive)|module|reexport|use(s|_module))(?=[(])', Keyword, 'root'),
-            (r'[a-z]\w*(?=[(])', Text, 'root'),
-            (r'[a-z]\w*[.]', Text, 'root'),
+            (r'[a-z][a-zA-Z0-9_]*(?=[(])', Text, 'root'),
+            (r'[a-z][a-zA-Z0-9_]*(?=[.])', Text, 'root'),
```
Comment (Anteru) [inline pygments/lexers/prolog.py]: Same as above -- is there a reason to not use `\w`?

Comment (pmoura) [reply pygments/lexers/prolog.py]: Same reason as above: avoid possible issues with Unicode code points.

File: pygments/lexers/prolog.py (hunk 2 above)
Comment (pmoura) [reply pygments/lexers/prolog.py]: Is my understanding that `\w` can include Unicode code points besides ASCII characters, which may result in an invalid atom.

File: pygments/lexers/prolog.py (hunk 1 above)
Comment (pmoura) [reply pygments/lexers/prolog.py]: No longer remember the motif. These changes are old (the original pull request was on BitBucket more than one year ago). But it works fine as I'm using this submitted version on a regular basis on a patched Pygments installation.

File: pygments/lexers/prolog.py (hunk 2 above)
Comment (Anteru) [reply pygments/lexers/prolog.py]: Ah yes, correct. I'm not sure we do unicode yet, but I guess if the language is limited to ASCII, doesn't hurt to make this explicit. Thanks!

File: pygments/lexers/prolog.py (hunk 1 above)
Comment (Anteru) [reply pygments/lexers/prolog.py]: Can you please give it a shot with `"''"` again? Other than that, should be ok to merge.

Comment (pmoura) [reply pygments/lexers/prolog.py]: Replace occurrences of `[']` with `'` and re-run the tests. No issues. Pushed the changes.

File: pygments/lexers/prolog.py (hunk 2 above)
Comment (birkenfeld) [reply pygments/lexers/prolog.py]: Well, Unicode is explicitly selected by re.UNICODE being in the flags for the lexer. If that's not accurate it can just be deleted.

File: pygments/lexers/prolog.py [hunk 4]
```diff
@@ -244,44 +246,43 @@ class LogtalkLexer(RegexLexer):
         'directive': [
             # Conditional compilation directives
             (r'(el)?if(?=[(])', Keyword, 'root'),
-            (r'(e(lse|ndif))[.]', Keyword, 'root'),
+            (r'(e(lse|ndif))(?=[.])', Keyword, 'root'),
```
Comment (birkenfeld) [inline pygments/lexers/prolog.py]: If a dot always follows, can it be highlighted in this rule?

Comment (pmoura) [reply pygments/lexers/prolog.py]: As you may know, Logtalk (and Prolog) source files are made of *terms* (directives and clauses; clauses can be facts or rules) that must always end with a dot. Thus, consistent highlighting would require to all ending dots be highlighted, not just in this particular case.

File: pygments/lexers/prolog.py (hunk 2 above)
Comment (pmoura) [reply pygments/lexers/prolog.py]: That explicit selection is done only in the Prolog lexer, not in the Logtalk lexer. Even there, is not clear. The ISO Prolog standards are mute on Unicode. Some Prolog systems support it, most don't. Among the Prolog systems that do support Unicode, there's no common de facto standard, particularly regarding using of non ASCII code points in identifiers.
